lib/api/1.1/northbound/profiles.js

```diff
@@ -4,7 +4,7 @@
 
 var di = require('di'),
     express = require('express'),
-    parser = require('body-parser');
+    ambiguous_parser = require('./../../../fittings/ambiguous_text_parser');
```

Comment:
I added a function to decide what parser should be used based on the content-type of the request... I had to do this because parser.raw() fails to achieve the desired result when using 'text/*' content-types, and parser.text() fails when you have 'application/octet-stream' content-type... so both were context specific. That being said, I would have had to feed this function in to both PUT definitions, both profiles.js and templates.js so it was suggested by @rolandpoulter to abstract this function call.